### ISSUE TITLE
Forced download link on MIME override file

### ISSUE THREAD

**Issue by Nbelles** [orhun/rustypaste#24]
Hello,

Is there a way to force the file to be downloaded when a file type has the MIME override enabled for that type? I want to be able to share the link to a file that would play in the browser (like .mp4) but also share a link that will auto-download as well (like with a flag in the URL or something similar). I thought I had seen this feature somewhere in this project but can't seem to find any documentation on it if it does exist. 

Thanks!

**Comment by orhun** [comment]
Hello!

This feature currently does not exist. You can only override the MIME types and file will be previewed or downloaded based on your browser settings. However, what you have suggested sounds interesting. Do you mean like a `?download=1` flag/parameter in the URL?

**Comment by Nbelles** [comment]
Gotcha. That's precisely what I'm talking about though. It would be nice to add that kind of flag to the URL to temporarily override any overrides (haha) that might be in place.

**Comment by orhun** [comment]
Hey, I implemented this in 99ac6156a8eafadee09fcba978f8353f4e9829ed

```sh
$ curl -s -o /dev/null -w '%{content_type}\n' $server/file.txt

text/plain; charset=utf-8
```

And when you specify `download=true`:

```sh
$ curl -s -o /dev/null -w '%{content_type}\n' $server/file.txt

application/octet-stream
```

This will force the browser to download the file.

Hope this helps! 🐻

**Comment by seqizz** [comment]
Just a random question: Is this expected to be cross-browser thing? On my test the file is downloaded on chrome but firefox insists on playing it: [example](https://paste.gurkan.in/hot-rattler.mp4?download=true)

**Comment by Nbelles** [comment]
I don't have Firefox but wanted to add that I have tested it working on Safari

**Comment by orhun** [comment]
> Just a random question: Is this expected to be cross-browser thing? On my test the file is downloaded on chrome but firefox insists on playing it: [example](https://paste.gurkan.in/hot-rattler.mp4?download=true)

Yeah I got the same result. The content type seems to be correct (`application/octet-stream`) so I'm not sure what is happening there.
